Userland/Applications/PixelPaint/Layer.cpp

```diff
@@ -90,6 +90,18 @@ void Layer::set_name(String name)
     m_image.layer_did_modify_properties({}, *this);
 }
 
+void Layer::flip(Gfx::Orientation orientation)
+{
+    m_bitmap = m_bitmap->flipped(orientation).release_nonnull();
+    did_modify_bitmap({});
+}
+
+void Layer::rotate(Gfx::RotationDirection rotation_direction)
+{
+    m_bitmap = m_bitmap->rotated(rotation_direction).release_nonnull();
```

Comment:
Same, but rotate in place.

Reply:
Maybe I'm missing or misunderstanding something, but rotating a non-squared bitmap in-place might be [complicated](https://en.wikipedia.org/wiki/In-place_matrix_transposition) and we might still have to use a secondary buffer, right ?

Reply:
I've discussed this matter with AtkinsSJ today and we didn't find a viable way to rotate a non-square bitmap in-place.
Maybe you have an idea in mind?

For now, I added the null-checking for the rotated() return value, and show an error on failure.
Maybe we could move that code to a Bitmap::rotate() method to have it alongside the Bitmap::flip() method I added.